Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchyBasedLinkResolver.swift

```diff
@@ -74,21 +74,38 @@ final class PathHierarchyBasedLinkResolver {
                 container.storage.compactMap { element in
                     guard let childID = element.node.identifier, // Don't include sparse nodes
                           !element.node.specialBehaviors.contains(.excludeFromAutomaticCuration),
-                          element.node.matches(languagesFilter: languagesFilter)
+                          element.node.matches(languagesFilter: languagesFilter),
+                          (element.node.overloadGroup == nil || element.node.overloadGroup?.identifier == node.identifier)
```

Comment:
If I understand this line correctly, it's excluding all the overloaded symbols except for the overload group itself. If so, this should add `.excludeFromAutomaticCuration` to the overloaded symbols instead.